packages/persona/src/Persona.purs

```diff
@@ -217,6 +224,13 @@ newtype SubscriptionState = SubscriptionState String
 derive instance genericSubscriptionState :: Generic SubscriptionState _
 instance readForeignSubscriptionState :: ReadForeign SubscriptionState where
   readImpl f = map SubscriptionState (readImpl f)
+instance eqSubscriptionState :: Eq SubscriptionState where
+  eq (SubscriptionState s1) (SubscriptionState s2) = s1 == s2
+instance ordSubscriptionState :: Ord SubscriptionState where
+  compare s1 s2
+    | isSubscriptionStateCanceled s1 = GT
+    | isSubscriptionStateCanceled s2 = LT
+    | otherwise = EQ
```

Comment:
This instance seems dodgy. It doesn't abides the laws either (see them [here](https://pursuit.purescript.org/packages/purescript-prelude/4.1.0/docs/Data.Ord#t:Ord)). If I understand correctly the intention was to make it so that cancelled subscription are always greater than others. I would have implemented it as:

```haskell
compare =
  comparing
    \s@(SubscriptionState st) ->
      if isSubscriptionStateCanceled s
      then Right st
      else Left st 
```

The idea here is that we defer all the real work to the existing `Ord (Either String String)` instance, so there can be no mistakes. We however implement the bit that we need by matching cancelled subscriptions and putting them as `Right` while everything else goes as `Left`, and [`Right` is always greater than `Left`](https://github.com/purescript/purescript-either/blob/v4.1.1/src/Data/Either.purs#L178)